core/src/replay_stage.rs

```diff
@@ -292,10 +293,23 @@ impl ReplayStage {
         blocktree: &Blocktree,
         progress: &mut HashMap<u64, ForkProgress>,
     ) -> Result<()> {
-        let (entries, num) = Self::load_blocktree_entries(bank, blocktree, progress)?;
-        let result = Self::replay_entries_into_bank(bank, entries, progress, num);
+        let result = {
+            let load_result = Self::load_blocktree_entries(bank, blocktree, progress);
+            if let Ok((entries, num)) = load_result {
+                Self::replay_entries_into_bank(bank, entries, progress, num)
+            } else {
+                // Just to appease the compiler so that the Result types match in both branches.
```

Comment:
you can't do a load_blocktree_entries.map() to get to the type you want?

Reply:
Yup, switched to using `and_then`, much cleaner :D